examples/aggregate_signature/registration.rs

```diff
@@ -0,0 +1,83 @@
+use blake2::digest::{Update, VariableOutput};
+use blake2::Blake2bVar;
+use blst::min_sig::PublicKey;
+use std::collections::BTreeMap;
+
+type Keys = BTreeMap<usize, PublicKey>;
+
+/// Structure for registration functionality.
+#[derive(Debug, Clone)]
+pub(crate) struct Registration {
+    /// Registered keys
+    pub(crate) registered_keys: Keys,
+    /// Checksum of registry data
+    pub(crate) checksum: Option<Vec<u8>>,
+}
+
+impl Registration {
+    /// Initialize key registration
+    pub(crate) fn new() -> Self {
+        Self {
+            registered_keys: BTreeMap::new(),
+            checksum: None,
+        }
+    }
+
+    /// Register a new key if registration is open and the key is not already registered.
+    pub(crate) fn register(&mut self, key: PublicKey, index: usize) -> bool {
+        if self.checksum.is_some() {
+            println!("Registration is closed!");
+            return false;
+        }
+
+        if self.registered_keys.values().any(|v| *v == key) {
+            println!("Key already registered!");
+            return false;
+        }
+
+        self.registered_keys.insert(index, key);
```

Comment:
You may be overwriting the key (if the index is the same for 2 parties)!

Reply:
I think it would be more natural for the register to output a Option<integer> where the integer is the signer's index (or none if already registered). 
You could instead check the length of the interator and return this as the index?